nautilus_trader/adapters/binance/http/market.py

```diff
@@ -853,15 +853,28 @@ async def request_binance_bars(
         end_time: Optional[str] = None,
     ) -> list[BinanceBar]:
         """Request Binance Bars from Klines."""
-        klines = await self.query_klines(
-            symbol=bar_type.instrument_id.symbol.value,
-            interval=interval,
-            limit=limit,
-            start_time=start_time,
-            end_time=end_time,
-        )
-        bars: list[BinanceBar] = [kline.parse_to_binance_bar(bar_type, ts_init) for kline in klines]
-        return bars
+        all_bars: list[BinanceBar] = []
+        while True:
+            klines = await self.query_klines(
+                symbol=bar_type.instrument_id.symbol.value,
+                interval=interval,
+                limit=limit,
+                start_time=start_time,
+                end_time=end_time,
+            )
+            bars: list[BinanceBar] = [
+                kline.parse_to_binance_bar(bar_type, ts_init) for kline in klines
+            ]
+            all_bars.extend(bars)
+
+            if len(klines) < limit:
```

Comment:
We want to keep track of how many bars we have received, and break out of the loop when either we reach the `limit` (if any), or there are no more klines to parse into bars.